local/modules/TheliaSmarty/Template/Plugins/DataAccessFunctions.php

```diff
@@ -325,10 +325,10 @@ public function cartDataAccess($params, $smarty)
                 break;
             case "total_price":
             case "total_price_with_discount":
-                $result = $cart->getTotalAmount();
+                $result = $cart->getTotalAmount(true, $taxCountry);
```

Comment:
As the state is also available through Thelia\TaxEngine\TaxEngine::getDeliveryState(), you ca use it in the call to $cart->getTotalAmount()